src/core/core.controller.js

```diff
@@ -53,19 +53,10 @@ module.exports = function(Chart) {
 			Chart.defaults.global,
 			Chart.defaults[chart.config.type],
 			newOptions);
-		// only reset scale(s) with options if options not reset
-		if (chart.options === chart.config.options) {
-			if (newOptions.scales) {
-				chart.options.scales = newOptions.scales;
-			} else if (newOptions.scale) {
-				chart.options.scale = newOptions.scale;
-			}
-		} else {
-			chart.options = newOptions;
-		}
-		chart.config.options = chart.options;
+
+		chart.options = chart.config.options = newOptions;
```

Comment:
Since a new object has already been created, it should be more clear to just reset the options, the result is actually the same.